modules/calib3d/src/solvepnp.cpp

```diff
@@ -723,4 +732,313 @@ void solvePnPRefineVVS(InputArray _objectPoints, InputArray _imagePoints,
+    _rvecs.create(solutions, 1, CV_MAKETYPE(depthRot, _rvecs.fixedType() && _rvecs.kind() == _InputArray::STD_VECTOR ? 3 : 1));
+    _tvecs.create(solutions, 1, CV_MAKETYPE(depthTrans, _tvecs.fixedType() && _tvecs.kind() == _InputArray::STD_VECTOR ? 3 : 1));
+
+    for (int i = 0; i < solutions; i++)
+    {
+        Mat rvec0, tvec0;
+        if (depthRot == CV_64F)
+            rvec0 = vec_rvecs[i];
+        else
+            vec_rvecs[i].convertTo(rvec0, depthRot);
+
+        if (depthTrans == CV_64F)
+            tvec0 = vec_tvecs[i];
+        else
+            vec_tvecs[i].convertTo(tvec0, depthTrans);
+
+        if (_rvecs.fixedType() && _rvecs.kind() == _InputArray::STD_VECTOR)
+        {
+            Mat rref = _rvecs.getMat_();
+
+            if (_rvecs.depth() == CV_32F)
+                rref.at<Vec3f>(0,i) = Vec3f(rvec0.at<float>(0,0), rvec0.at<float>(1,0), rvec0.at<float>(2,0));
+            else
+                rref.at<Vec3d>(0,i) = Vec3d(rvec0.at<double>(0,0), rvec0.at<double>(1,0), rvec0.at<double>(2,0));
+        }
+        else
+        {
+            _rvecs.getMatRef(i) = rvec0;
+        }
+
+        if (_tvecs.fixedType() && _tvecs.kind() == _InputArray::STD_VECTOR)
+        {
+
+            Mat tref = _tvecs.getMat_();
+
+            if (_tvecs.depth() == CV_32F)
+                tref.at<Vec3f>(0,i) = Vec3f(tvec0.at<float>(0,0), tvec0.at<float>(1,0), tvec0.at<float>(2,0));
+            else
+                tref.at<Vec3d>(0,i) = Vec3d(tvec0.at<double>(0,0), tvec0.at<double>(1,0), tvec0.at<double>(2,0));
+        }
+        else
+        {
+            _tvecs.getMatRef(i) = tvec0;
+        }
+    }
+
+    if (reprojectionError.needed())
+    {
+        int type = reprojectionError.type();
+        reprojectionError.create(solutions, 1, type);
+        CV_CheckType(reprojectionError.type(), type == CV_32FC1 || type == CV_64FC1,
+                     "Type of reprojectionError must be CV_32FC1 or CV_64FC1!");
+
+        Mat objectPoints, imagePoints;
+        if (_opoints.depth() == CV_32F)
+        {
+            _opoints.getMat().convertTo(objectPoints, CV_64F);
+        }
+        else
+        {
+            objectPoints = _opoints.getMat();
+        }
+        if (_ipoints.depth() == CV_32F)
+        {
+            _ipoints.getMat().convertTo(imagePoints, CV_64F);
+        }
+        else
+        {
+            imagePoints = _ipoints.getMat();
+        }
+
+        for (size_t i = 0; i < vec_rvecs.size(); i++)
+        {
+            vector<Point2d> projectedPoints;
+            projectPoints(objectPoints, vec_rvecs[i], vec_tvecs[i], cameraMatrix, distCoeffs, projectedPoints);
+
+            Mat projectedPoints_(projectedPoints);
+            projectedPoints_ = projectedPoints_.reshape(1, static_cast<int>(projectedPoints.size()*2));
+            imagePoints = imagePoints.reshape(1, static_cast<int>(projectedPoints.size()*2));
+            Mat rmse = projectedPoints_ - imagePoints;
+            rmse = rmse.t() * rmse;
```

Comment:
Last 4 lines: `norm(projectedPoints, imagePoints, NORM_L2);`

`NORM_L2 | NORM_RELATIVE` - for `rmse_`.

Reply:
I did not manage to get the same results and I don't think it is the [same formula](https://docs.opencv.org/4.1.0/d2/de8/group__core__array.html#gad12cefbcb5291cf958a85b4b67b6149f).
For RMSE, I use:

![image](https://user-images.githubusercontent.com/8229425/57873337-383e6800-780e-11e9-895a-b9f35fbf0a6a.png)

whereas in the doc:

![image](https://user-images.githubusercontent.com/8229425/57873417-7dfb3080-780e-11e9-92da-c3448790e7fb.png)

Reply:
the lines: 
```
Mat rmse = projectedPoints_ - imagePoints;
rmse = rmse.t() * rmse;
```
are misleading - they definitely do not comptue RMSE according to its standard definition

Reply:
You are right!

```
rmse_ = norm(projectedPoints_, imagePoints, NORM_L2) / sqrt(projectedPoints_.total());
```
?

Reply:
I think it should be 
`rmse_ = sqrt(norm(projectedPoints_, imagePoints, NORM_L2) / projectedPoints_.total());`